src/core/core/model/story.py

```diff
@@ -918,7 +999,42 @@ def update_stories(cls, stories: set["Story"]):
                 logger.exception(f"Update Story: {story.id} Failed")
 
     @classmethod
-    def create_from_item(cls, news_item):
+    def prepare_misp_stories(cls, story_lists: list[dict], force) -> list[dict]:
+        stories = []
+        for story in story_lists:
+            if existing_story := cls.get(story.get("id", {})):
+                if isinstance(existing_story, Story):
+                    if not force and cls.check_internal_changes(existing_story.to_detail_dict()):
```

Comment:
Why not only get stories where internal is True instead of getting every story and checking for internal afterwards.